From bfd80332b5f4804244f14d475bc0182bd0553a64 Mon Sep 17 00:00:00 2001
Message-Id: <bfd80332b5f4804244f14d475bc0182bd0553a64.1531783043.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 17 Jul 2018 01:15:30 +0200
Subject: [PATCH] meson: Fix xkbcommon-x11.pc Requires

The meson-generated pkgconfig file was missing Requires and
Requires.private.

Signed-off-by: Jan Alexander Steffens (heftig) <jan.steffens@gmail.com>
---
 meson.build | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 7ee4f78..1243eae 100644
--- a/meson.build
+++ b/meson.build
@@ -216,48 +216,52 @@ if get_option('enable-x11')
 You can disable X11 support with -Denable-x11=false.''')
     endif
 
+    libxkbcommon_x11_deps = [
+        xcb_dep,
+        xcb_xkb_dep,
+    ]
+
     libxkbcommon_x11_internal = static_library(
         'xkbcommon-x11-internal',
         'src/x11/keymap.c',
         'src/x11/state.c',
         'src/x11/util.c',
         'src/x11/x11-priv.h',
         'src/context.h',
         'src/context-priv.c',
         'src/keymap.h',
         'src/keymap-priv.c',
         'src/atom.h',
         'src/atom.c',
         include_directories: include_directories('src'),
         link_with: libxkbcommon,
-        dependencies: [
-            xcb_dep,
-            xcb_xkb_dep,
-        ],
+        dependencies: libxkbcommon_x11_deps,
     )
     libxkbcommon_x11_link_args = []
     if have_version_script
         libxkbcommon_x11_link_args += '-Wl,--version-script=' + join_paths(meson.source_root(), 'xkbcommon-x11.map')
     endif
     libxkbcommon_x11 = library(
         'xkbcommon-x11',
         'xkbcommon/xkbcommon-x11.h',
         link_whole: libxkbcommon_x11_internal,
         link_args: libxkbcommon_x11_link_args,
         link_depends: 'xkbcommon-x11.map',
         version: '0.0.0',
         install: true,
     )
     install_headers(
         'xkbcommon/xkbcommon-x11.h',
         subdir: 'xkbcommon',
     )
     pkgconfig.generate(
         name: 'xkbcommon-x11',
         filebase: 'xkbcommon-x11',
         libraries: libxkbcommon_x11,
         version: meson.project_version(),
         description: 'XKB API common to servers and clients - X11 support',
+        requires: libxkbcommon,
+        requires_private: libxkbcommon_x11_deps,
     )
 endif
 
-- 
2.18.0

